### PR TITLE
RichTextBlockItem

### PR DESCRIPTION
**Pull request by praspra2**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented the RichTextBlock for the model based approach.
- https://app.vidcast.io/share/77f778cb-c89b-4d90-804a-fc4d758d6613

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml_v2/AdaptiveCardQmlEngine/utils/Utils.h [hunk 1]
```diff
@@ -16,13 +16,31 @@ namespace AdaptiveCardQmlEngine
         static const std::string handleEscapeSequences(std::string& text);
         static const std::string formatHtmlUrl(std::string& text, const std::string& linkColor, const std::string& textDecoration);
         static std::vector<std::string> splitString(const std::string& string, char delimiter);
+        static std::string& replace(std::string& str, char what, char with);
+        //static std::string& replace(std::string& str, const std::string& what, const std::string& with);
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/utils/Utils.h:21]: remove commented code.

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp [hunk 2]
```diff
@@ -23,20 +22,21 @@ namespace AdaptiveCardQmlEngine
...
+    // ReInitializing AdaptiveCard and Host config
         mCardConfig = std::make_shared<AdaptiveCardConfig>(mAdaptiveCardTheme == AdaptiveCardEnums::AdaptiveCardTheme::DarkTheme ? true : false);
 
         if (mAdaptiveCardTheme == AdaptiveCardEnums::AdaptiveCardTheme::DarkTheme)
         {
-			mHostConfig = std::make_shared<AdaptiveCards::HostConfig>(AdaptiveCards::HostConfig::DeserializeFromString(DarkConfig::darkConfig));
-		}
+            mHostConfig =
+            std::make_shared<AdaptiveCards::HostConfig>(AdaptiveCards::HostConfig::DeserializeFromString(DarkConfig::darkConfig));
+        }
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp:32]: Please make corrections in indentation.

**Review by soumykun — COMMENTED** [review]
Please address the minor comments.

**Review by soumykun — APPROVED** [review]

**Review by soumykun — APPROVED** [review]

File: source/qml_v2/AdaptiveCardQmlEngine/qml/RichTextBlockRender_old.qml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.3
+import QtQuick.Controls 2.15
+import AdaptiveCardQmlEngine 1.0
+import "JSUtils/AdaptiveCardUtils.js" as AdaptiveCardUtils
+
+TextEdit {
```
Comment (ramkrishnajana) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/RichTextBlockRender_old.qml:7]: What is this `_old` file?

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp [hunk 4]
```diff
@@ -14,7 +13,7 @@ namespace AdaptiveCardQmlEngine
 
     void AdaptiveCardContext::initAdaptiveCardContext()
     {
-        // Initializing Host config and Card config
+    // Initializing Host config and Card config
```
Comment (ramkrishnajana) [inline source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp:16]: indentation

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp [hunk 5]
```diff
@@ -23,20 +22,19 @@ namespace AdaptiveCardQmlEngine
     {
         mAdaptiveCardTheme = theme;
 
-        // ReInitializing AdaptiveCard and Host config
+    // ReInitializing AdaptiveCard and Host config
```
Comment (ramkrishnajana) [inline source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp:25]: indentation

Comment (praspra2) [reply source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp]: Done

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp (hunk 4 above)
Comment (praspra2) [reply source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardContext.cpp]: Done

File: source/qml_v2/AdaptiveCardQmlEngine/qml/RichTextBlockRender_old.qml (hunk 3 above)
Comment (praspra2) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/RichTextBlockRender_old.qml]: This was a duplicate file of RichTextBlock.qml which was created by mistake and is not required so i have removed this in the latest commit changes.

**Pull request merged**